vm/src/stdlib/array.rs

```diff
@@ -431,6 +431,25 @@ impl PyArray {
         }
     }
 
+    #[pymethod(name = "__ne__")]
+    fn ne(lhs: PyObjectRef, rhs: PyObjectRef, vm: &VirtualMachine) -> PyResult {
```

Comment:
Instead of copying `__eq__` implementation, refactoring `__eq__` to use `PyComparisonValue` like `bytes` will allow this method reuse the idiomatic `__ne__` implementation.

Reply:
Similar work: https://github.com/RustPython/RustPython/pull/2092/commits/feb1b895d7ce5fc8b4368c69b4ea65d0916037a8#diff-c9329f7f730f2866bea2eb2f3eebb95cR129-R145